### PR TITLE
BaSP-MR-47: async, await and fix issues POST, PUT and DELETE

### PR DESCRIPTION
**Pull request by ukiweskamp**

### CODE REVIEW

**Review by Federico-Peralta — APPROVED** [review]
🚀

File: src/controllers/trainer.js [hunk 1]
```diff
@@ -45,43 +62,47 @@ const updateTrainer = (req, res) => {
...
+      return res.status(404).json({
+        message: `The trainer with the ID: ${id} doesn't exists`,
+        data: undefined,
+        error: true,
       });
-    })
-    .catch((error) => res.status(400).json({
-      message: 'An error ocurred',
+    }
+
```
Comment (iaracriscenti) [inline src/controllers/trainer.js:75]: Could you check if the email or dni entered already exists? Like u did on the post controller

File: src/controllers/trainer.js [hunk 2]
```diff
@@ -92,74 +113,80 @@ const deleteTrainer = (req, res) => {
...
-        message: 'Trainers list',
-        data: trainer,
-        error: false,
+const getAllTrainer = async (req, res) => {
+  try {
+    const trainers = await Trainer.find();
+    if (!trainers) {
+      return res.status(404).json({
+        message: 'Trainers not found',
+        error: true,
```
Comment (iaracriscenti) [inline src/controllers/trainer.js:147]: please add data: undefined

File: src/controllers/trainer.js [hunk 3]
```diff
@@ -92,74 +113,80 @@ const deleteTrainer = (req, res) => {
...
-    })
-    .catch((error) => res.status(400).json({
+    }
+    return res.status(200).json({
+      message: 'Trainers list',
+      data: trainers,
+      error: false,
+    });
+  } catch (error) {
+    return res.status(400).json({
```
Comment (iaracriscenti) [inline src/controllers/trainer.js:156]: Same here, data value should be undefined and error should be a boolean

**Review by iaracriscenti — CHANGES_REQUESTED** [review]
Great job! Please check every response structure to match in every case.

File: src/controllers/trainer.js [hunk 4]
```diff
@@ -185,6 +200,7 @@ const getTrainerById = async (req, res) => {
   } catch (error) {
     return res.status(400).json({
       message: `An error occurred to find the trainer with ID ${id}`,
+      data: undefined,
       error,
```
Comment (JulianRiedinger7) [inline src/controllers/trainer.js:204]: this error should also be a boolean.. true in this case

File: src/controllers/trainer.js (hunk 1 above)
Comment (iaracriscenti) [reply src/controllers/trainer.js:86]: Great addition! But this validation is still missing. I meant something like:
```
  const found = await Trainer.findOne({ $or: [{ dni }, { email }] });

    if (found) {
      return res.status(400).json({
        message: `The admin with DNI ${dni} or Email ${email} already exists`,
        data: undefined,
        error: true,
      });
    }
```

**Review by iaracriscenti — APPROVED** [review]

**Pull request merged**